core/unit_test/tstRemove.hpp

```diff
@@ -25,40 +25,45 @@ void testRemove()
     int num_particle = 200;
     Cabana::AoSoA<Cabana::MemberTypes<int>, TEST_MEMSPACE> aosoa(
         "remove", num_particle );
-    // Purposely using zero-init here.
-    Kokkos::View<int*, TEST_MEMSPACE> keep_particle( "keep", num_particle );
-
     auto keep_slice = Cabana::slice<0>( aosoa, "slice" );
+
+    // Remove nothing
+    Kokkos::parallel_for(
+        "init", Kokkos::RangePolicy<TEST_EXECSPACE>( 0, num_particle ),
+        KOKKOS_LAMBDA( const int p ) { keep_slice( p ) = 1; } );
+
+    Cabana::remove( TEST_EXECSPACE{}, num_particle, keep_slice, aosoa );
+    EXPECT_EQ( aosoa.size(), num_particle );
+
+    // Remove only odd particles.
+    aosoa.resize( num_particle );
+    keep_slice = Cabana::slice<0>( aosoa, "slice" );
+
     Kokkos::parallel_for(
         "init", Kokkos::RangePolicy<TEST_EXECSPACE>( 0, num_particle ),
         KOKKOS_LAMBDA( const int p ) {
             if ( p % 2 )
             {
                 keep_slice( p ) = 1;
-                keep_particle( p ) = 1;
             }
             else
             {
                 keep_slice( p ) = 0;
             }
         } );
 
-    // Remove only odd particles.
     int new_num_particle = num_particle / 2;
-    Cabana::remove( TEST_EXECSPACE{}, new_num_particle, keep_particle, aosoa );
+    Cabana::remove( TEST_EXECSPACE{}, new_num_particle, keep_slice, aosoa );
     EXPECT_EQ( aosoa.size(), new_num_particle );
```

Comment:
Can you add one more check that each particle left after the removal has a value of 1 (that we removed the right particles)? That was intended to be included in the original version